Raise `ActiveRecord:: InverseOfAssociationNotFoundError` if invalid inverse_of is specified (#51767)
rails/rails: activerecord/CHANGELOG.md

*   Raise `ActiveRecord::InverseOfAssociationNotBidirectionalError` if invalid inverse_of is specified

>     The association specified in `inverse_of` should be the name of the association
>     on the other model that points back to the current model.
>     If the user inadvertently specifies an invalid association name in `inverse_of`
>     that is not bi-directional, the `inverse_of` is implicitly ignored.
>     However, if it remains implicitly ignored, the user can't know that the inverse_of
>     is incorrect. Therefore, I have made a change to raise an exception instead of.

Reviewer: ```suggestion
    This error occurs when the association specified in the `inverse_of` option does not exist on the associated class.
    Previously this would be implicitly ignored, so the developer wouldn't know they tried to make an invalid association.
```

    ```ruby
    Post.belongs_to(:user, inverse_of: :comment) #
    user = User.create!

     # Before:
    Post.new(user: user) #=> No error

     # After:

>     Post.new(user: user) #=> ActiveRecord::InverseOfAssociationNotBidirectionalError: Inverse association user (:comment in User) is not bi-directional.

Reviewer: ```suggestion
    Post.new(user: user)
    #=> ActiveRecord::InverseOfAssociationNotBidirectionalError: Inverse association user (:comment in User) is not bi-directional.
```

    ```

> 
>     When checking the bi-directionality of inverse_of reflection, it is intuitive to
>     compare classes. However, the implementation compares table_name instead, because
>     class name comparisons are challenging with anonymous classes (such as those used in Rails testing).

Reviewer: ```suggestion
```

This is implementation detail which should not go into release notes.
